flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-iceberg/src/main/java/org/apache/flink/cdc/connectors/iceberg/sink/v2/compaction/CompactionOperator.java

```diff
@@ -0,0 +1,151 @@
+
+    private Catalog catalog;
+
+    /** store a list of MultiTableCommittable in one checkpoint. */
+    private final Map<TableId, Integer> tableCommitTimes;
+
+    private final Set<TableId> compactedTables;
+
+    private final CompactionOptions compactionOptions;
+
+    private volatile Throwable throwable;
+
+    private ExecutorService compactExecutor;
+
+    public CompactionOperator(
+            Map<String, String> catalogOptions, CompactionOptions compactionOptions) {
+        this.tableCommitTimes = new HashMap<>();
+        this.compactedTables = new HashSet<>();
+        this.catalogOptions = catalogOptions;
+        this.compactionOptions = compactionOptions;
+    }
+
+    @Override
+    public void open() throws Exception {
+        super.open();
+        if (compactExecutor == null) {
+            this.compactExecutor =
+                    Executors.newSingleThreadScheduledExecutor(
+                            new ExecutorThreadFactory(
+                                    Thread.currentThread().getName() + "-Cdc-Compaction"));
+        }
+    }
+
+    @Override
+    public void processElement(StreamRecord<CommittableMessage<WriteResultWrapper>> element) {
+        if (element.getValue() instanceof CommittableWithLineage) {
+            TableId tableId =
+                    ((CommittableWithLineage<WriteResultWrapper>) element.getValue())
+                            .getCommittable()
+                            .getTableId();
+            tableCommitTimes.put(tableId, tableCommitTimes.getOrDefault(tableId, 0) + 1);
+            int commitTimes = tableCommitTimes.get(tableId);
+            if (commitTimes >= compactionOptions.getCommitInterval()
+                    && !compactedTables.contains(tableId)) {
+                if (throwable != null) {
+                    throw new RuntimeException(throwable);
+                }
+                compactedTables.add(tableId);
+                if (compactExecutor == null) {
+                    compact(tableId);
+                } else {
+                    compactExecutor.submit(() -> compact(tableId));
+                }
+            }
+        }
+    }
+
+    private void compact(TableId tableId) {
+        if (catalog == null) {
+            catalog =
+                    CatalogUtil.buildIcebergCatalog(
+                            "cdc-iceberg-compaction-catalog", catalogOptions, new Configuration());
+        }
+        try {
+            RewriteDataFilesActionResult rewriteResult =
+                    Actions.forTable(
+                                    StreamExecutionEnvironment.createLocalEnvironment(),
+                                    catalog.loadTable(TableIdentifier.parse(tableId.identifier())))
+                            .rewriteDataFiles()
+                            .execute();
+            LOGGER.info(
+                    "Iceberg small file compact result for {}: {} data files and {} delete files",
```

Comment:
Iceberg small file compact result for {}: added {} data files and deleted {} files.